src/sqlfluff/testing/rules.py

```diff
@@ -193,6 +197,15 @@ def rules__test_helper(test_case):
             assert res == test_case.fix_str
             if test_case.violations_after_fix:
                 assert_violations_after_fix(test_case)
+            else:
+                assert_rule_pass_in_sql(
+                    test_case.rule,
+                    test_case.fix_str,
+                    configs=test_case.configs,
+                    msg="No violations_after_fix was provided, but the SQL after fix"
+                    "violates the rule. To accept a partial fix, "
+                    "violations_after_fix must be set.",
```

Comment:
This reads a little confusing. WDYT about this?:

```suggestion
                    msg="The SQL after fix is applied still contains rule violations. "
                    "To accept a partial fix, violations_after_fix must be set "
                    "listing the remaining, expected, violations.",
```